### ISSUE TITLE
Copying query results to the clipboard does not always work

### ISSUE THREAD

**Issue by sqlsauce** [microsoft/azuredatastudio#1388]
Issue Type: <b>Bug</b>

Steps to repro:

1. On Windows 7, open SQL Operations Studio.
2. Run a SQL query (ad-hoc or stored procedure) that returns results.
3. Select and copy any part or all of the results using Ctrl-C or right-click and either Copy or Copy With Headers.

Expected result: query results are copied to the clipboard.

Actual result: query results are not copied to the clipboard and cannot be pasted.

SQL Operations Studio version: sqlops 0.29.3 (7099922c35bc3b3e223228c972cc03cee5abbe25, 2018-05-04T05:46:24.032Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4810MQ CPU @ 2.80GHz (8 x 2793)|
|Memory (System)|7.88GB (1.07GB free)|
|Process Argv|C:\Program Files\SQL Operations Studio\sqlops.exe|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (3)</summary>

Extension|Author (truncated)|Version
---|---|---
agent|Mic|0.29.0
server-report|Mic|0.1.2
whoisactive|Mic|0.1.1

</details>
Reproduces without extensions

**Comment by nilamo** [comment]
I wish I knew how to describe this better.  But it's like one out of four attempts to copy to the clipboard actually do as expected.  Most of the time, it seems like nothing happened at all.  

Version 0.32.8
Commit efd470b3c771ef7473ba3cf108a33b6579c96a53
Date 2018-08-30T19:40:40.895Z
VS Code 1.23.1
Shell 1.7.16
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by aaomidi** [comment]
Can someone confirm that they're noticing this recently too? There's been a lot of changes made to the way copy/paste works and would be interested to see if this problem still exists.

**Comment by aaomidi** [comment]
Since there hasn't been any updates recently, I'll close this issue.

Please update or create a new issue with any specific issues you're seeing with the clipboard.

**Comment by marcselman** [comment]
I am still seeing this issue.

What I notice is that the data gets copied, but sometimes it takes seconds.
So if I copy then paste immediately I sometimes paste my old clipboard value.
If I then wait a couple of seconds the new value is available in the clipboard.

I know there has been a fix released but even with the latest version I'm still experiencing this.

**Comment by aaomidi** [comment]
Is the data larger than normal by any chance? Curious to see if there is some latency in copying to clipboard.

**Comment by marcselman** [comment]
No, just an integer from a single cell for example.

**Comment by chengkiang** [comment]
I am getting this randomly on the Mac version. Sometimes, the copy would work, other times, it wouldn't. E.g. If I copy row 3 column 2, I can paste. Now, if I need row 5 col 6, I'd highlight that, Cmd-C, and when I Cmd-V, I am still getting the earlier piece that I copied (Row 3 Col 2). It's very annoying.

**Comment by jelitter** [comment]
Why was this closed when it's clearly not fixed ?

- When selecting a cell in results and pressing **Ctrl + A**, results lose focus and query gets selected instead.
- **Ctrl + C** does not copy selected cells.
- **Right Click** in results does not show a context menu.

**Comment by dnaren** [comment]
> * When selecting a cell in results and pressing **Ctrl + A**, results lose focus and query gets selected instead.
> * **Ctrl + C** does not copy selected cells.
> * **Right Click** in results does not show a context menu.

I was running into similar issues, even with the latest version of ADS. I was able to resolve my issues by deleting the %AppData%\azuredatastudio folder (effectively a factory reset of ADS). 

PS: You'll need to re-configure your connections & any preferences you had before.

**Comment by jelitter** [comment]
> > * When selecting a cell in results and pressing **Ctrl + A**, results lose focus and query gets selected instead.
> > * **Ctrl + C** does not copy selected cells.
> > * **Right Click** in results does not show a context menu.
> 
> I was running into similar issues, even with the latest version of ADS. I was able to resolve my issues by deleting the %AppData%\azuredatastudio folder (effectively a factory reset of ADS).
> 
> PS: You'll need to re-configure your connections & any preferences you had before.

Thanks very much @dnaren , that worked 👍

**Comment by bbirtle** [comment]
I'm having the issue described, with exactly the same steps to reproduce. It seems intermittent, but very frequent. In some (all?) cases, the clipboard finally does get the expected contents only after a long delay (and no progress/activity indication).

I tried also saving to Excel and in that case, a popup window appears and takes ages to finally save the file.

Perhaps is this related to the other very major issue that connections to SQL Server take an insanely long time (also very frequently)? Although I haven't ruled out that issue could be something to do with my particular server and unrelated to the Azure client software.

If you need further details or I should open a new bug report please let me know. I'm using the latest version of macOS and connecting in this case to a server running on Windows Server 2019 and SQL Server 2019, with encryption enabled, Trust Server Certificate enabled, and SQL Login (via a private VPN relay to avoid man in middle attacks under this configuration).
 
I will in the meantime try @jelitter 's solution (and other workarounds).

TIA! 🙏 😃

**Comment by plmosqueda** [comment]
Hi, im having this issue on Linux Debian and Fedora too :-(

**Comment by Charles-Gagnon** [comment]
@plmosqueda Can you try the latest [insiders](https://github.com/Microsoft/azuredatastudio#try-out-the-latest-insiders-build-from-main) build? We recently fixed an issue with copy on Linux which will be included in the next release - https://github.com/microsoft/azuredatastudio/pull/24341

**Comment by plmosqueda** [comment]
@Charles-Gagnon Thank you man. Works really good on Debian 12. I will try later o Fedora.

**Comment by rick4096** [comment]
This issue still happens. If you click Results -> Save Results -> Copy to clipboard, nothing gets copied to the clipboard, and the previous buffer in the clipboard is what gets pasted if you try to Cmd+V.

I'm using Safari on macOS. This is super annoying. Currently I am copying results out as JSON, and using a 3rd-party website to render that as an HTML table which I can then copy-paste. Please help.
